snapcraft.yaml

```diff
@@ -382,10 +466,8 @@ parts:
       # Same goes for arm64 builds on Launchpad.  It seems we can't
       # detect $GITHUB_WORKSPACE, so we assume presence of the file
       # "symbols-upload-token" means we are running on GitHub.
-      if [ -f "$CRAFT_PROJECT_DIR/symbols-upload-token"] || [ $CRAFT_ARCH_BUILD_FOR = "arm64" ] || [ $CRAFT_ARCH_BUILD_FOR = "armhf" ]; then
+      if [ $CRAFT_ARCH_BUILD_ON = "armhf" ] || [ $CRAFT_ARCH_BUILD_ON = "arm64" ]; then
```

Comment:
The comment needs to be updated, as it does not mention armhf and mentions symbols-upload-token.

If I understand correctly, this is bound with the replacement of the line below: `if [ $CRAFT_ARCH_BUILD_FOR = "amd64" ] || [ $CRAFT_ARCH_BUILD_FOR = "arm64" ]; then` -> `if [ $CRAFT_ARCH_BUILD_ON = "amd64" ]; then`.

All two parts do not relate to the cross-compilation so they would ideally be in a separate commit of their own, explaining that we now want this in arm64 too for $REASONS.

Reply:
Actually I think it's fine to just drop the mention to it from the comments above this line.

Reply:
@nteodosio question is more: is this still needed on launchpad hardware? does a build without this patch applied succeeds?

Reply:
Yes, c.f. https://launchpad.net/~nteodosio/+snap/firefox-cross-disabled.


But I'm just noticing that https://github.com/lissyx/firefox-snap/blob/snapcraft_cross_compile_gnome-46-2404_NO_SYSROOT/snapcraft.yaml#L808-L815 will also needs clean-up then.